packages/expo-video/src/VideoPlayer.types.ts

```diff
@@ -180,6 +181,26 @@ export declare class VideoPlayer extends SharedObject<VideoPlayerEvents> {
    * @platform ios
    */
   generateThumbnailsAsync(times: number | number[]): Promise<VideoThumbnail[]>;
+
+  // We re-declare some of the functions from the event emitter to explicitly hide them in the docs without adding
+  // expo-video specific exceptions in the docs code.
+  /**
+   * @hidden
+   */
+  emit<EventName extends keyof VideoPlayerEvents>(
+    eventName: EventName,
+    ...args: Parameters<VideoPlayerEvents[EventName]>
+  ): void;
+
+  /**
+   * @hidden
+   */
+  startObserving?<EventName extends keyof VideoPlayerEvents>(eventName: EventName): void;
+
+  /**
+   * @hidden
+   */
+  stopObserving?<EventName extends keyof VideoPlayerEvents>(eventName: EventName): void;
```

Comment:
Please, don't do this 😅 We're already trying to hide inherited methods in the docs, see https://github.com/expo/expo/blob/b86f1fe71efe421b2ebeb152d5b59ebd554e14cb/docs/components/plugins/api/APISectionClasses.tsx#L43-L48
Maybe there is an issue in that logic.

Reply:
The issue is that `child.inheritedFrom` which is compared to `CLASSES_TO_IGNORE_INHERITED_PROPS ` does not take into account nested inheritance. So if we inherit some methods from `VideoPlayer` (for exaple play()) but `VideoPlayer` also inherits other methods from `EventEmitter` (e.g. `addEventListener`), the docs always list the inheritance as from `VideoPlayer` for both methods, therefore there is no way to display the inherited of `VideoPlayer` methods without showing the `EvenEmitter` methods. It would be doable if we could do `child.inheritedFrom.inheritedFrom.inheritedFrom ...` to check the entire inheritance chain. 

@Simek Do you know if something like this would be doable before the SDK 52 release?

Reply:
Thanks for the investigation! 🙏  The answer for now is "maybe", not sure tho, since the data comes from TypeDoc. The question is if we can mutate the data in a wanted way or improve the parsing/logic around it. Can look at this tomorrow

Reply:
@behenate For now, let's add `'NativeVideoModule.VideoPlayer'` entry in the array that Tomasz has mention, it should solve the problem.

It looks like export structure confuses TypeDoc due to exporting `VideoPlayer` class directly and then avoiding `VideoPlayer` type export in index, but making it exportable type and using it in the analysed code. Look like this leads to merge of type and class, and results in multiple entries for some children in final entity, which have different `inheritedFrom` values.